cmd/bugsnag/release.go

```diff
@@ -0,0 +1,184 @@
+		),
+
+		releaseStage: releaseCmd.String(
+			"release-stage",
+			"production",
+			`Optional. The environment in which your application is running.
+Set to "" explicitly to report a build instead of a release.`,
+		),
+
+		provider: releaseCmd.String(
+			"provider",
+			"",
+			`Optional. Your version control provider.
+Valid values: "github", "github-enterprise", "bitbucket", "bitbucket-server", "gitlab", "gitlab-onpremise"`,
+		),
+
+		repository: releaseCmd.String(
+			"repository",
+			"",
+			`Optional, unless revision is also set. URL of your repository.`,
+		),
+
+		revision: releaseCmd.String(
+			"revision",
+			"",
+			`Optional, unless repository is also set.
+The SHA (or 7-character shorthand) of the git commit associated with the build.`,
+		),
+
+		builder: releaseCmd.String(
+			"builder",
+			"",
+			`Optional. The name of the person or bot performing this release.`,
+		),
+
+		metadata: releaseCmd.String(
+			"metadata",
+			"",
+			`Optional. Format is "KEY1=VALUE1,KEY2=VALUE2"`,
+		),
+
+		autoAssignRelease: releaseCmd.Bool(
+			"auto-assign-release",
+			false,
+			`Optional. Set to true if the logic in your application is unaware of its own app version.`,
+		),
+
+		endpoint: releaseCmd.String(
+			"endpoint",
+			"",
+			`Optional. If you're running Bugsnag on-prem, set this to the build API's URL`,
+		),
+
+		appBundleVersion: releaseCmd.String(
+			"app-bundle-version",
+			"",
+			`Optional. Applies to Apple platform builds only.`,
+		),
+
+		appVersionCode: releaseCmd.Int(
+			"app-version-code",
+			0,
+			`Optional. Applies to Android builds only.`,
+		),
+
+		debug: releaseCmd.Bool("debug", false, "Turn on for debug logs"),
+	}
+}
+
+func (app *application) runRelease(envVars map[string]string) error {
+	rf := app.releaseFlags
+	if *rf.debug {
+		app.printReleaseDebug()
+	}
+
+	req := &builds.JSONBuildRequest{
+		APIKey:            *rf.apiKey,
+		AppVersion:        *rf.appVersion,
+		ReleaseStage:      *rf.releaseStage,
+		BuilderName:       *rf.builder,
+		Metadata:          splitByEquals(strings.Split(*rf.metadata, ",")),
+		AppVersionCode:    *rf.appVersionCode,
+		AppBundleVersion:  *rf.appBundleVersion,
+		AutoAssignRelease: *rf.autoAssignRelease,
+	}
+
+	if *rf.repository != "" || *rf.revision != "" {
+		req.SourceControl = &builds.JSONSourceControl{
+			Provider:   *rf.provider,
+			Repository: *rf.repository,
+			Revision:   *rf.revision,
+		}
+	}
+
+	populateReleaseDefaults(req, envVars)
+
+	if err := req.Validate(); err != nil {
+		return fmt.Errorf("Invalid build data: %w\nSee 'bugsnag release --help'", err)
+	}
+
+	publisher := builds.DefaultPublisher()
+	if endpoint := *rf.endpoint; endpoint != "" {
+		publisher = builds.NewPublisher(endpoint)
+	}
+	if err := publisher.Publish(req); err != nil {
+		return err
+	}
+
+	fmt.Printf("release info published for version %s\n", req.AppVersion)
+	return nil
+}
+
+func (app *application) printReleaseDebug() {
```

Comment:
Should probably request that users invoke and include the output of `bugsnag release` with the  `--debug` flag before raising any bug reports.